src/bucket/Bucket.h

```diff
@@ -109,10 +115,10 @@ class Bucket : public std::enable_shared_from_this<Bucket>,
     // stored with their corresponding liquidity pool key in
     // liquidityPoolKeyToTrustline. All liquidity pool keys corresponding to
     // loaded trustlines are also reduntantly stored in liquidityPoolKeys.
-    // If a trustline key is in deadTrustlines, it is not loaded. Whenever a
-    // dead trustline is found, its key is added to deadTrustlines.
+    // If a trustline key is in seenTrustlines, it is not loaded. Whenever a
+    // dead trustline is found, its key is added to seenTrustlines.
```

Comment:
dead->seen? I don't quite understand the logic of the change here though.

Reply:
For a given key, there is only one valid version of the entry. However, because the bucket list is a log-structured merge tree, multiple different versions of a key can exist simultaneously on different levels. For example, if a key has been recently deleted, there may be a DEADENTRY of the key marking that it has been deleted on level 0, but a LIVEENTRY of the same key on some lower level N. For this particular query, we have to scan all levels of the BucketList. Previously, this set was called `deadTrustlines` because it kept track of all the DEADENTRY keys we had already seen. That way, if we scan a DEADENTRY on level 0, we know not to later load that entry if we find a LIVEENTRY version on a lower level.

However, if an entry has been updated multiple times, it's possible that there are multiple LIVEENTRIES on multiple levels of the bucket list with different values of the same key. Suppose we have some key k that exists as a LIVEENTRY on both level 0 and 2. This function would first load the newest version on level 0, and emplace it into `liquidityPoolKeyToTrustline`. Later on, we then load the outdated version on level 2. This key is not in `deadTrustlines`, since there has been no DEADENTRY. So, we would again call `emplace` into `liquidityPoolKeyToTrustline` for the outdated version of the entry. In practice this is fine, since the key already exists in the map so emplace would just be a no-op.

This change cleans up the code to make this behavior more explicit. The first time we see a key, we add it to `seenTrustlines`. Whenever we call emplace, we check to make sure we haven't seen the key yet. If we have already seen the key, we know the version we just scanned is outdated, so we don't call emplace and continue.

Reply:
Thanks for the explanation!